runtime/autoload/health/provider.vim

```diff
@@ -106,7 +106,14 @@ endfunction
 
 " Fetch the contents of a URL.
 function! s:download(url) abort
-  if executable('curl')
+  if executable('curl') && !empty(
+        \ filter(
+        \   split(
+        \     join(
+        \       filter(systemlist(['curl', '-V']), 'v:val =~# "^Protocols:\\s\\+"'),
+        \     ' '),
+        \   ' '),
+        \ 'v:val =~? "^https$"'))
```

Comment:
What message will be given to the user? With the old code we already show "curl failed ...". So if we now show "missing curl", that doesn't really improve anything.

Reply:
And maybe it's good enough to just use the shorter and more readable:

```vim
system('curl -V') =~# 'Protocols:.*https'
```